### PR TITLE
Improved the unique validation rule test cases

### PR DESCRIPTION
**Pull request by LukeTowers**
Resolves some inconsistencies noticed by @Flynsarmy.

@bennothommo @Flynsarmy if you guys could do a sanity check on this for me please that would be great.

### CODE REVIEW

File: src/Database/Traits/Validation.php [hunk 1]
```diff
@@ -435,31 +436,42 @@ protected function processRuleFieldNames($rules)
...
+        $params = ["unique:{$connection}.{$table}", $column, $ignoreValue, $ignoreColumn];
 
-        if ($whereColumn) {
+        if ($whereColumn && $whereValue) {
             $params[] = $whereColumn;
-        }
-
-        if ($whereValue) {
             $params[] = $whereValue;
         }
```
Comment (bennothommo) [inline src/Database/Traits/Validation.php]: @LukeTowers would this break conditions that don't have a value - not sure if it's possible, but I wonder if this method previously supported things like "ISNULL(deleted_at)", which technically does not need a value to be used as a condition.

Comment (LukeTowers) [reply src/Database/Traits/Validation.php]: @bennothommo good catch, looks like that is supported in the Unique rule object at least: https://github.com/laravel/framework/blob/8.x/src/Illuminate/Validation/Rules/DatabaseRule.php#L93-L95. Mind adding a test for it and pushing the fix?

File: src/Database/Traits/Validation.php [hunk 2]
```diff
@@ -435,31 +436,42 @@ protected function processRuleFieldNames($rules)
...
             $whereValue
         ) = array_pad(explode(',', $definition, 6), 6, null);
 
-        $table = 'unique:' . $this->getConnectionName()  . '.' . $this->getTable();
-        $column = $column ?: $fieldName;
-        $key = $keyName ? $this->$keyName : $this->getKey();
-        if (is_null($key)) {
-            $key = 'NULL';
+        // Remove unique or unique: from the table name
+        $table = ltrim(ltrim($table, 'unique'), ':');
```
Comment (LukeTowers) [inline src/Database/Traits/Validation.php]:
```suggestion
        $table = Str::after(Str::after($table, 'unique'), ':');
```

**Comment by LukeTowers** [comment]
@bennothommo is this good to go?

**Comment by bennothommo** [comment]
@LukeTowers yep, I believe so. :)

**Pull request merged**